content/posts/jspi.md

```diff
@@ -0,0 +1,419 @@
+---
+title: "WebAssembly stack switching and the sync async problem"
+date: 2025-05-13T13:33:51-04:00
+draft: true
+tags: ["announcement"]
+# author: "Me"
+# author: ["Me", "You"] # multiple authors
+showToc: true
+TocOpen: false
+draft: false
+hidemeta: false
+comments: false
+description: "Desc Text."
+# canonicalURL: "https://canonical.url/to/page"
+disableHLJS: true # to disable highlightjs
+disableShare: false
+hideSummary: false
+searchHidden: true
+ShowReadingTime: true
+ShowBreadCrumbs: true
+ShowPostNavLinks: true
+cover:
+    # image: "<image path/url>" # image path/url
+    # alt: "<alt text>" # alt text
+    # caption: "<text>" # display caption under cover
+    relative: false # when using page bundles set this to true
+    hidden: true # only hide on current single page
+---
+
+Writing Python code in the browser, we constantly run into what I call the
+sync/async problem. Namely, most Python code and most libc functions are
+synchronous, but most JavaScript APIs are asynchronous. For example, `urllib`
+offers a synchronous API to make a request, but in JavaScript, the normal way to
+make a request is with the `fetch` API which is asynchronous. Python also has
+APIs like `event_loop.run_until_complete()` which are supposed to block until an
+async task is completed. There was simply no way to do this.
+
+JavaScript Promise integration is a new WebAssembly standard that gives us a way
+to work around this. It allows us to make a call that seems synchronous from
+the perspective of C but is actually asynchronous from the perspective of
+Python.
```

Comment:
perspective of Python or perspective of JavaScript? (actually both seems the same anyways)

Reply:
I meant to say JavaScript here indeed.